### PR TITLE
fix(ddtrace-run): drop distutils in favor of shutil

### PR DESCRIPTION
**Pull request by jd**
The current code calls `distutils.spawn.find_executable` on the passed
argument. This is fine on non-Windows platform, but it fails easily on Windows
platform.

`find_executable` has a subtle bug reported as
https://github.com/python/cpython/issues/94338 which prevents files with a
`.EXE` extension (uppercase) to be found.

As `distutils` is deprecated, replace it with shutil which is not and has a
`which` function which does the same thing.

### CODE REVIEW

**Review by Kyle-Verhoog — DISMISSED** [review]

**Comment by Kyle-Verhoog** [comment]
release note? 🤷‍♂️

File: ddtrace/commands/ddtrace_run.py [hunk 1]
```diff
@@ -101,8 +101,11 @@ def main():
         sys.exit(1)
 
     # Find the executable path
-    executable = spawn.find_executable(args.command[0])
-    if not executable:
+    if os.path.isfile(args.command[0]):
+        executable = args.command[0]
+    else:
+        executable = spawn.find_executable(args.command[0])
```
Comment (P403n1x87) [inline ddtrace/commands/ddtrace_run.py:107]:
```suggestion
    executable = args.command[0]
    if not os.path.isfile(executable)
        executable = spawn.find_executable(executable)
```

**Comment by Kyle-Verhoog** [comment]
> @Kyle-Verhoog release note for a non-user-visible change? 🤔

It would be a pretty significant bug/fix on windows, right?

**Review by Kyle-Verhoog — APPROVED** [review]

**Comment by jd** [comment]
> It would be a pretty significant bug/fix on windows, right?

Not sure TBH. It's only a bug if you pass `.EXE` as an extension which seems to be not that often.
(but happened to me on the CI 🤷)

**Review by brettlangdon — APPROVED** [review]

**Review by Kyle-Verhoog — DISMISSED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

File: ddtrace/commands/ddtrace_run.py [hunk 2]
```diff
@@ -14,6 +16,23 @@
...
+    _which = spawn.find_executable  # type: ignore[assignment]
+
+
+def find_executable(
+    p,  # type: str
+):
+    # type: (...) -> typing.Optional[str]
+    if os.path.isfile(p):
+        return p
+    return _which(p)
```
Comment (P403n1x87) [inline ddtrace/commands/ddtrace_run.py:34]: nit 
```suggestion
    return p if os.path.isfile(p) else _which(p)
```

**Pull request merged**